mtls-go-apple-keychain/signer/keychain.go

```diff
@@ -23,92 +23,94 @@ import (
 const (
 	// TLS cipher suites: https://www.rfc-editor.org/rfc/rfc8446.html#section-9.1
 	supportedAlgorithm = tls.PSSWithSHA256
-	commonName         = "testClientTLS"
 	maxCertificatesNum = 10
 )
 
-func GetClientCertificate(info *tls.CertificateRequestInfo) (*tls.Certificate, error) {
-	fmt.Printf("Server requested certificate\n")
+func GetClientCertificate(commonName string) func(*tls.CertificateRequestInfo) (*tls.Certificate, error) {
+	getClientCert := func(info *tls.CertificateRequestInfo) (*tls.Certificate, error) {
```

Comment:
there's no change here except 

- `commonName` is no longer a global const in this package, it's scope is only inside this function
- `GetClientCertificate` returns `func(*tls.CertificateRequestInfo) (*tls.Certificate, error)`

all the rest of the code is the same